lib/spark.js

```diff
@@ -418,3 +481,64 @@ function mapErrorToStatusCode(err) {
   // Fallback code for unknown errors
   return 600
 }
+
+/**
+ * Picks a random provider based on their priority and supplied randomness.
+ *
+ * Providers are prioritized in the following order:
+ *
+ * 1. HTTP Providers with Piece Info advertised in their ContextID.
+ * 2. Graphsync Providers with Piece Info advertised in their ContextID.
+ * 3. HTTP Providers.
+ * 4. Graphsync Providers.
+ *
+ * @param {Provider[]} providers
+ * @param {number} randomness
+ * @returns {Provider | undefined}
+ */
+export function pickRandomProvider(providers, randomness) {
+  const rng = new Prando(randomness)
+
+  const filterByProtocol = (items, protocol) =>
+    items.filter((provider) => provider.protocol === protocol)
+
+  const pickRandomItem = (items) => {
+    if (!items.length) return undefined
+    return items[Math.floor(rng.next() * items.length)]
```

Comment:
How have we decided that the `Math.random()` pseudo number generator isn't enough?

Reply:
We have discussed on using RNG with seed in order to get deterministic alternative providers in order to make committee generation possible. As I am aware it's not possible to provide seed to `Math.random()`, rather it uses seed generated by underlying OS. That's why I have opted for not using `Math.random()` pseudo RNG.

Reply:
got it!